content/blog/2025-05-08-go-client-GA.md

```diff
@@ -0,0 +1,503 @@
+The design provides a natural, Go-like interface for transactions through:
+
+- Clean command calling syntax (`cmd.Set(...)`, `cmd.Get(...)`)
+- Use of Go's embedding to inherit behavior from `baseClient`
+- Explicit error handling that follows Go conventions (`result, err := tx.Exec()`)
+
+### Command Transparency
+
+A key design followed is that commands can be used the same way in both regular and transaction contexts:
+
+```go
+// In regular client context - executes immediately
+client.Set("key1", "value1")
+
+// In transaction context - identical syntax, but queued instead
+tx := NewTransaction(client)
+cmd := tx.GlideClient
+cmd.Set("key1", "value1")  // Queued for later execution
+```
+
+- Regular clients and transactions both implement a common interface
+- When creating a transaction, it sets itself as the command executor
+- In transaction mode, commands are queued instead of executed
+- The code detects transaction context and handles results appropriately
+
+### Optimistic Concurrency Support
+
+Valkey's optimistic locking mechanism is supported through:
+
+- Implementation of `Watch()` that monitors keys for changes
+- Implementation of `Unwatch()` that releases monitoring
+- Proper handling of transaction failure when watched keys change
+- Passing watch state through to the Valkey server during execution
+
+This enables patterns like:
+
+```go
+cmd.Watch([]string{"key1"})
+// Check current value
+val, _ := cmd.Get("key1")
+// Only update if no one else modified since Watch
+cmd.Set("key1", "new-value")
+// Will fail if key1 changed between Watch and Exec
+result, err := tx.Exec()
+```
+
+### Discard Operation
+
+ The `tx.Discard()` method can be used to explicitly cancel a transaction before execution.
+
+This mechanism serves to:
+
+- Release server-side resources associated with the transaction when the client decides not to proceed. - Clean up client-side state, such as the command queue within the `TransactionExecutor`.
+- Prevent the execution of unintended commands.
+ This functionality is achieved by sending the `DISCARD` command to the Valkey server, which then clears the server-side command queue for the transaction. The `tx.Discard()` method will return an error if communication with the server fails during this process.
+
+## Join the Journey
+
+We're actively developing and enhancing the Go client, and we'd love your feedback and contributions. Try it out in your projects, share your experiences, and help us make it even better!
+You can join our development journey by:
+
+- Submitting issues or feature requests on our [GitHub Issues page](https://github.com/valkey-io/valkey-glide/issues)
+- Joining discussions in our [GitHub Discussions forum](https://github.com/valkey-io/valkey-glide/discussions)
+
+## Looking Forward
+
+As we move toward general availability, we'll be expanding command support, enhancing performance, and adding even more features to make the Valkey GLIDE Go client a great choice for Go developers.
```

Comment:
- "As we move toward general availability"... wait - isn't this announcing GA not moving towards it.
- "expanding command support" you say earlier that you've got all the commands: which is it?